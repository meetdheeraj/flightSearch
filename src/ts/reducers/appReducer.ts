import { Reducer } from 'redux';
import * as object_assign from 'object-assign';

import { IFlightAppState } from '../StoreTypes';
import { IFlightAppAction } from '../actions/AppStoreActions';

export const initialState: IFlightAppState = {
  date: new Date().toString(),
  city: {
    fromCity: 'Pune',
    toCity: 'Bangalore'
  },
  flightList: [
    {
      city: {
        fromCity: 'Pune',
        toCity: 'Bangalore'
      },
      price: 1145,
      date: new Date().toString()
    }
  ]
};

export function AppReducer(
  state: IFlightAppState,
  action: IFlightAppAction
): IFlightAppState {
  if (typeof state === 'undefined') {
    return initialState;
  }

  let newState: IFlightAppState = {
    ...state
  };

  switch (action.type) {
    case 'SET_DATE':
      newState = {
        ...state,
        date: action.date
      };

      return object_assign({}, state, newState);

    case 'SET_CITY':
      newState = {
        ...state,
        city: action.city
      };

      return object_assign({}, state, newState);

    case 'SET_FLIGHT_LIST':
      newState = {
        ...state,
        flightList: action.flightList
      };

      return object_assign({}, state, newState);
  }

  return state;
}
