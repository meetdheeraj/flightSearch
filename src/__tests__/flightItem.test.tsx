import * as React from 'react';
import { createStore } from 'redux';
import { AppReducer } from '../ts/reducers/appReducer';
import { Provider } from 'react-redux';
import Flight from '../ts/components/flightItem';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';

describe('<Flight />', () => {
    it('should render successfully', () => {
        const store = createStore(AppReducer);
        const renderer = new ReactShallowRenderer();

        const flightItem = {
            city: {
                fromCity: 'Pune',
                toCity: 'Bangalore'
            },
            price: 4456,
            date: new Date().toString()
        }
        expect(renderer.render(
            <Provider store={store}><Flight flightItem={flightItem} /></Provider>
        )).toMatchSnapshot()
    })
});
