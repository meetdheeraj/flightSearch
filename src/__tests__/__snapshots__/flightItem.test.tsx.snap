// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flight /> should render successfully 1`] = `
<Flight
  flightItem={
    Object {
      "city": Object {
        "fromCity": "Pune",
        "toCity": "Bangalore",
      },
      "date": "Mon Dec 10 2018 17:51:20 GMT+0530 (IST)",
      "price": 4456,
    }
  }
/>
`;
